lib/bitswap.go

```diff
@@ -69,6 +74,14 @@ func CheckBitswapCID(ctx context.Context, c cid.Cid, ma multiaddr.Multiaddr, get
 		return nil, err
 	}
 
+	tctx, cancel := context.WithTimeout(ctx, time.Second*10)
+	defer cancel()
+	// Create a new stream to ensure hole punching happens
+	_, err = h.NewStream(tctx, ai.ID, "/ipfs/bitswap/1.2.0", "/ipfs/bitswap/1.1.0", "/ipfs/bitswap/1.0.0", "/ipfs/bitswap")
+	if err != nil {
+		return nil, err
+	}
```

Comment:
We should be clearer with the comment here so we know when we can remove this. IIUC it's more like:
```suggestion
	// Create a new stream to ensure we wait for hole punching even if it takes longer than the built-in limit in the Bitswap implementation
	_, err = h.NewStream(tctx, ai.ID, "/ipfs/bitswap/1.2.0", "/ipfs/bitswap/1.1.0", "/ipfs/bitswap/1.0.0", "/ipfs/bitswap")
	if err != nil {
		return nil, err
	}
```

Ideally also with a link to two issues:
- In boxo for the making the limit configurable
- In go-libp2p for holepunching taking far too long (which I know you're debugging with @sukunrt anyway)